Tile.py

```diff
@@ -2,19 +2,49 @@
 
 
 class Tile:
-    def __init__(self, x, y, row, col, isObstacle=False, isBloated=False, isKnown=False):
+    top_terabee, mid_terabee, bottom_terabee, lidar = 0, 1, 2, 3
+
+    def __init__(self, x, y, row, col, isObstacle = False, isBloated = False, isKnown=False):
         """
         Initialize a tile centered at x coordinate [x] and y coordinate [y].
         If [isObstacle] is True, the tile is initialized as an obstacle, else Tile is
         marked as free space, [isObstacle] is False by default.
         """
+        self.obstacle_score = [0, 0, 0, 0]
+        self.bloat_score = 0
+        self.bloat_tiles = []
+        self.is_found = False
         self.x = x
         self.y = y
         self.row = row
         self.col = col
         self.is_obstacle = isObstacle
         self.is_bloated = isBloated
 
+    def increase_score(self, sensor_type):
+        """
+        increase the score at that sensor type, up to a certain bound (maximal score)
+        If this score fits certain conditions, then update the is_obstacle boolean accordingly
+        """
+        self.obstacle_score[sensor_type] = min(obstacle_threshold, self.obstacle_score[sensor_type]+1)
+        if self.obstacle_score[sensor_type]==obstacle_threshold:
+            self.is_obstacle = True
+
+    def decrease_score(self, sensor_type):
+        """
+        decreases the score at that sensor type, as low as 0
+        If this score fits certain conditions, then update the is_obstacle boolean accordingly
+        """
+        self.obstacle_score[sensor_type] = max(0, self.obstacle_score[sensor_type] - 1)
```

Comment:
@kimmylin727 @kennycai0 Make the amount you decrease by an argument to the function or from consts.py

Reply:
resolving